Tasks/Npm/npm.ts

```diff
@@ -30,6 +33,58 @@ async function main(): Promise<void> {
     }
 }
 
+async function _logNpmStartupVariables() {
+    try {
+        // Log the NPM version
+        let version: string;
+        try {
+            let syncResult: IExecSyncResult = tl.execSync('npm', '--version');
+            if (syncResult.stdout) {
+                version = syncResult.stdout.trim();
+            }
+        } catch (err) {
+            tl.debug(`Unable to get NPM config info. Err:( ${err} )`);
+        }
+
+        // Log the NPM registries
+        let command = tl.getInput(NpmTaskInput.Command);
+        let npmRegistriesAry: INpmRegistry[];
+        let registryUrlAry = [];
+        switch (command) {
+            case NpmCommand.Install:
+            case NpmCommand.Custom:
+                npmRegistriesAry = await npmCustom.getCustomRegistries();
```

Comment:
This is now executed twice during a build, could we make it so it only happens one and log the telemetry at the end of the task? Getting the registries currently invokes a call to the location service to get the location for local feeds, which I think some customers have seen issues in the past. Would like to avoid having to make those calls twice. We could also revisit the logic to try and avoid making those calls.

Reply:
See above on "exit logs".  I hear you and didn't like adding this twice.  I'll see what I can do to avoid that call.  Maybe set a global var or something.